CometServer/Services/Operations/ModelCreatorManager.cs

```diff
@@ -255,21 +269,21 @@ public void CopyEngineeringModelData(EngineeringModelSetup newModelSetup, Npgsql
 
             var sw = Stopwatch.StartNew();
             this.Logger.LogDebug("start modify {Count} references of things contained in the new engineering-model-setup (rdl included)", modelThings.Count);
+
             foreach (var modelThing in modelThings)
             {
-                var model = modelThing as EngineeringModel;
-                if (model != null)
+                if (modelThing is EngineeringModel model)
                 {
                     continue;
                 }
 
-                var iteration = modelThing as Iteration;
-                if (iteration != null)
+                if (modelThing is Iteration iteration)
```

Comment:
Since you remove all iteration that are not the active one, can not you filter here that it is the Active Iteration there? So you can keep the InvalidOperationException later?

Reply:
No, because we don't have that information here. We don't know the reason why the IterationSetup is not in this.originalToCopyMap. Complete flow of ModelCreatorManager needs to change to make sure we can handle this, otherwise we will create horrible spaghetti code.